milabench/validation/validation.py

```diff
@@ -0,0 +1,158 @@
+import json
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+
+from ..structs import BenchLogEntry
+
+
+class ValidationLayer:
+    """Validation layer interface, captures events, makes report"""
+
+    _return_code = None
+
+    def __init__(self, **kwargs) -> None:
+        # early stop means voir requested milabench to stop the benchmark
+        # this means we can ignore the process return code because it got SIGTERM'ed
+        self.early_stop = False
+
+        # return code of the validation layer
+        # this is used to make milabench fail on critical errors
+        self._return_code = None
+
+    def __call__(self, entry):
+        return self._on_event(entry)
+
+    def _on_event(self, entry: BenchLogEntry):
+        self.early_stop = False
+
+        if entry.event == "stop":
+            self.early_stop = True
```

Comment:
That could be the body of the default implementation of `on_stop`.

Reply:
done